tutorial/practicals/LFRic/building_code/4_psydata/README.md

```diff
@@ -0,0 +1,205 @@
+# Example 4: Using PSyData Transformations
+
+In this example you will use some of the available PSyData
+transformation to instrument the previous LFRic example.
+Initially we will be using kernel extraction as an example,
+but a list of other PSyData applications is provided at the end.
+
+
+## Step 1: Compile all required PSyData libraries
+Change directory to PSyclone's ``lib`` directory, and trigger
+compilation of all PSyData libraries.
+
+    cd $PSYCLONEHOME/lib
+    make all
+
+You need the NetCDF development package installed, the makefiles
+will be using ``nf-config`` to get the appropriate compiler and
+linker flags. Additionally, compiler and compiler flags can be provided
+using the environment variables F90 and F90FLAGS:
+
+    F90=ifort F90flags="-O2 -traceback" make all
+
+By default gfortran will be used.
+
+
+## Step 2: Create a transformation script
+This transformation script will be used as a parameter
+to PSyclone when building the application. The script will apply
+transformation to insert
+code to extract input- and output-data of a kernel. Initially
+we will only transform the kernel in ``time_evolution_alg_mod.x90``.
+Use the simple script ``transform_one.py`` as a template
+to create a new transformation script that applies the
+LFRic-specific kernel extraction transformation. The transformation
+can be imported from ``psyclone.domain.lfric.transformations``
+and is called ``LFRicExtractTrans``.
+
+The ``apply()`` function in this script is called by PSyclone before
+any code is created. Inside ``apply`` create an instance of the
+extraction transformation and apply it to the schedule for
+"invoke_propagate_perturbation". The template file ``transform_one.py``
+contains most of the required calls, you only need to fill in the
```

Comment:
You might need a bit more hand-holding here - especially since this is the first Python we get them to do.

Reply:
Hmm - I thought long about this, but I can't really say what else I can do (short of giving them files where the right solution is commented out). I've added as intro where to find the solution, and I will actually show this kind of code on my slides, so they can also always go back to them.

Reply:
I suggest getting them to look at the `invoke` in the `time_evolution_alg_mod.x90` that you supply in order to find the name to use for the `psy.invokes.get()` call.

Reply:
You are right. I have split the first part of the instructions into three substeps as suggested. This seems indeed to be easier to understand now (though I really don't like that we have to add 'invoke_' to the name ... but we have #994 for this).